ctapipe/instrument/camera.py

```diff
@@ -518,6 +533,69 @@ def get_shower_coordinates(self, x, y, psi):
 
         return longi, trans
 
+    def get_pixel_id(self, x, y):
+        '''
+        Return the camera pixel number which contains a given position (x,y) 
+        in the camera frame. The (x,y) coordinates can be arrays (of equal length),
+        for which the methods returns an array of pixel ids. A warning is raised if the 
+        position falls outside the camera.
+
+        Parameters
+        ----------
+        x: astropy.units.Quantity (distance) of horizontal position(s) in the camera frame
+        y: astropy.units.Quantity (distance) of vertical position(s) in the camera frame
+
+        Returns
+        -------
+        pixel_ids: Pixel number or array of pixel numbers. Returns -1 if position falls 
+                   outside camera
+        '''
+
+        if not np.all(self.pix_area == self.pix_area[0], axis=0):
+            logger.warning(" Method not implemented for cameras with varying pixel sizes")
+
+        points_searched = np.dstack([x.to_value(u.m), y.to_value(u.m)])
+
+        kdtree = self.all_pixels_kdtree
+        dist, pixel_ids = kdtree.query(points_searched)
+        pixel_ids = pixel_ids.flatten()
+
+        # Check if the position lies inside the camera. It is first checked if any border
+        # pixel numbers are returned. If not, everything is fine. If yes, the distance of 
+        # the given position to the closest pixel center is translated to the distance to 
+        # the center of a non-border pixel', pos -> pos', and it is checked whether pos' 
+        # still lies within pixel'. If not, pos lies outside the camera. This approach  
+        # does not need to know the particular pixel shape, but as the kdtree itself, 
+        # presumes all camera pixels being of equal size.
+
+        border_mask = self.get_border_pixel_mask()
+        # get all pixels at camera border:
+        borderpix_ids = np.where(border_mask)[0]
+
+        borderpix_ids_in_list = np.intersect1d(borderpix_ids, pixel_ids)
+        if borderpix_ids_in_list.any():
+            # Get some pixel not at the border:
+            inside_pix_id = np.where(~border_mask)[0][0]
+            # Check in detail whether location is in border pixel or outside camera:
+            for borderpix_id in borderpix_ids_in_list:
+                index = np.where(pixel_ids == borderpix_id)[0][0]
+                # compare with inside pixel:
+                xprime = points_searched[0][index, 0] \
```

Comment:
for calculations, use implicit line continuations in parenthese instead of explicit line continuations, like this:

```
a = (
    b 
    + c
)
```

Reply:
Also, converting units is kind of expensive, so you should do `self.pix_y[index].to_value(u.m)` instead of converting all values and then only using one.

Reply:
done in fd5d48f.